src/VisualStudio/CSharp/Impl/Options/Formatting/CodeStylePage.cs

```diff
@@ -11,7 +19,502 @@ internal class CodeStylePage : AbstractOptionPage
+
+            editorconfig.AppendLine();
+            editorconfig.AppendLine("# C# files");
+            editorconfig.AppendLine("[*.cs]");
+            editorconfig.Append("indent_style = ");
+            editorconfig.AppendLine(CSharpCodeStyleOptions_GenerateEditorconfig(optionSet, FormattingOptions.UseTabs));
+            editorconfig.Append("indent_size = ");
+            editorconfig.AppendLine(optionSet.GetOption(FormattingOptions.IndentationSize, LanguageNames.CSharp).ToString());
+            editorconfig.Append("insert_final_newline = ");
+            editorconfig.AppendLine(optionSet.GetOption(FormattingOptions.InsertFinalNewLine).ToString().ToLower());
+
+            editorconfig.AppendLine();
+            // .NET Coding Conventions
+            editorconfig.AppendLine("###############################");
+            editorconfig.AppendLine("# .NET Coding Conventions     #");
+            editorconfig.AppendLine("###############################");
+            editorconfig.AppendLine("# Organize usings");
+            editorconfig.Append("dotnet_sort_system_directives_first = ");
+            editorconfig.AppendLine(optionSet.GetOption(GenerationOptions.PlaceSystemNamespaceFirst, LanguageNames.CSharp).ToString().ToLower());
+
+            editorconfig.AppendLine();
+            // this. preferences
+            editorconfig.AppendLine("# this. preferences");
+            editorconfig.Append("dotnet_style_qualification_for_field = ");
+            editorconfig.AppendLine(CSharpCodeStyleOptions_GenerateEditorconfig(optionSet, CodeStyleOptions.QualifyFieldAccess));
+            editorconfig.Append("dotnet_style_qualification_for_property = ");
+            editorconfig.AppendLine(CSharpCodeStyleOptions_GenerateEditorconfig(optionSet, CodeStyleOptions.QualifyPropertyAccess));
+            editorconfig.Append("dotnet_style_qualification_for_method = ");
+            editorconfig.AppendLine(CSharpCodeStyleOptions_GenerateEditorconfig(optionSet, CodeStyleOptions.QualifyMethodAccess));
+            editorconfig.Append("dotnet_style_qualification_for_event = ");
+            editorconfig.AppendLine(CSharpCodeStyleOptions_GenerateEditorconfig(optionSet, CodeStyleOptions.QualifyPropertyAccess));
+
+            editorconfig.AppendLine();
+            // Language keywords vs. BCL types preferences
+            editorconfig.AppendLine("# Language keywords vs BCL types preferences");
+            editorconfig.Append("dotnet_style_predefined_type_for_locals_parameters_members = ");
+            editorconfig.AppendLine(CSharpCodeStyleOptions_GenerateEditorconfig(optionSet, CodeStyleOptions.PreferIntrinsicPredefinedTypeKeywordInDeclaration));
+            editorconfig.Append("dotnet_style_predefined_type_for_member_access = ");
+            editorconfig.AppendLine(CSharpCodeStyleOptions_GenerateEditorconfig(optionSet, CodeStyleOptions.PreferIntrinsicPredefinedTypeKeywordInMemberAccess));
+
+            editorconfig.AppendLine();
```

Comment:
can you either keep the AppendLines with the preceding group, or put the // comment above hte Appendline.